app/scripts/services/planner.service.js

```diff
@@ -66,18 +66,34 @@
             // check to see if we've finished all the bosses
             if (step.bosses) {
                 angular.forEach(step.bosses, function(boss) {
-                    if (boss.isAlliance && items.isAlliance && items.lookup[boss.ID] === undefined) {
-                        neededBosses.push(boss);
-                        completed = false;
+
+                    var bossIsNeutral = !boss.isAlliance && !boss.isHorde,
+                        character = items, // aliasing for clarity
+                        characterNeedsBoss = function(id){ return !character.lookup[id]; },
+                        addBoss = function(boss) {
+                            neededBosses.push(boss);
+                            completed = false;
+                        };
```

Comment:
I think the style of the codebase is to just use multiple `var` instead of using commas

Reply:
Yup, fixed.